test/repl.jl

```diff
@@ -397,6 +397,64 @@ begin
     # Try entering search mode while in custom repl mode
     LineEdit.enter_search(s, custom_histp, true)
 end
+
+# Test removal of prompt in bracket pasting
+begin
+    stdin_write, stdout_read, stderr_read, repl = fake_repl()
+
+    repl.interface = REPL.setup_interface(repl)
+    repl_mode = repl.interface.modes[1]
+    shell_mode = repl.interface.modes[2]
+    help_mode = repl.interface.modes[3]
+
+    repltask = @async begin
+        Base.REPL.run_repl(repl)
+    end
+
+    c = Condition()
+
+    sendrepl2(cmd) = write(stdin_write,"$cmd\n notify(c)\n")
+
+    # Test single statement
+    sendrepl2("\e[200~julia> A = 2\e[201~\n")
+    wait(c)
+    @test A == 2
+
+    # Test multi statement
+    sendrepl2("""\e[200~
+            julia> type T17599; a::Int; end
+
+                function foo(julia)
+            julia> 3
+                end
+
+                    julia> A = 3\e[201~
+             """)
+    wait(c)
+    @test A == 3
+    @test foo(4)
+    @test T17599(3).a == 3
+    @test !foo(2)
+
+    # Test only works in bracket mode
+    sendrepl2("julia = 4\n julia> 3 && (A = 1)\n")
+    wait(c)
+    @test A == 1
+
+    # Test shell> mode
+    tmpdir = mktempdir()
```

Comment:
I don't see many other tests in this file creating or deleting things? If there were and they weren't using do blocks yet I'd go fix them (in a separate PR if needed).